File: src/main/java/network/kinau/recipes/entities/Recipe.java
package network.kinau.recipes.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.Hibernate;

import java.util.Date;
import java.util.LinkedHashSet;
import java.util.Objects;
import java.util.Set;

@Entity
@Table(name = "recipe")
@Getter
@Setter
@ToString
@RequiredArgsConstructor
public class Recipe {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "description", nullable = false)
    private String description;

    @Column(name = "image", nullable = false)
    private String image;

    @Column(name = "servings", nullable = false)
    private Long servings;

    @Column(name = "time", nullable = false)
    private Long time;

    @Column(name = "created_at", nullable = true)
    private Date createdAt;

    @OneToMany(mappedBy = "recipe")
    @ToString.Exclude
    private Set<RecipeIngredient> recipeIngredients = new LinkedHashSet<>();

    @OneToMany(mappedBy = "recipe")
    @ToString.Exclude
    private Set<Step> steps = new LinkedHashSet<>();

    @ManyToMany
    @JoinTable(name = "recipe_tag",
            joinColumns = @JoinColumn(name = "recipe_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id"))
    private Set<Tag> tags = new LinkedHashSet<>();

    @ManyToMany
    @JoinTable(name = "recipe_category",
            joinColumns = @JoinColumn(name = "recipe_id"),
            inverseJoinColumns = @JoinColumn(name = "category_id"))
    private Set<Category> categories = new LinkedHashSet<>();

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Recipe recipe = (Recipe) o;
        return id != null && Objects.equals(id, recipe.id);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: Frontend/src/models/Recipe.ts
import Ingredient from "@/models/Ingredient";
import Step from "@/models/Step";
import Tag from "@/models/Tag";

export default class Recipe {
    id: string;
    name: string;
    description: string;
    ingredients: Ingredient[];
    steps: Step[];
    image: string;
    time: number;
    servings: number;
    tags: Tag[];
    createdAt: string;


    constructor(id: string, name: string, description: string, ingredients: Ingredient[], steps: Step[], image: string, time: number, servings: number, tags: Tag[], createdAt: string) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.ingredients = ingredients;
        this.steps = steps;
        this.image = image;
        this.time = time;
        this.servings = servings;
        console.log(tags);
        this.tags = tags;
        this.createdAt = createdAt;
    }
}